middleware/oauth.global.ts

```diff
@@ -24,7 +24,78 @@ export default defineNuxtRouteMiddleware(async (to) => {
       }
     }
   }
+  console.log("to.path", to.path);
+  // Check if this is a dataset route that might have public access
+  // Check if this is a dataset route that might have public access
+  const isDatasetRoute =
+    to.path.startsWith("/alerts/") ||
+    to.path.startsWith("/gallery/") ||
+    to.path.startsWith("/map/");
 
+  if (isDatasetRoute) {
+    console.log("isDatasetRoute", isDatasetRoute, "for path:", to.path);
+    try {
+      const {
+        public: { appApiKey },
+      } = useRuntimeConfig();
+      const headers = { "x-api-key": appApiKey };
+
+      const data = (await $fetch("/api/config", { headers })) as unknown as [
+        { [key: string]: { routeLevelPermission?: RouteLevelPermission } },
+        string[],
+      ];
+
+      // Extract table name from path (e.g., "/alerts/tableName" -> "tableName")
+      const pathParts = to.path.split("/");
+      const tableName = pathParts[pathParts.length - 1];
+
+      const viewConfig = data?.[0]?.[tableName];
+      const permission: RouteLevelPermission =
+        viewConfig?.routeLevelPermission ?? "anyone"; // Default to anyone if not set
+
+      // If view is accessible to anyone, allow access without authentication
+      if (permission === "anyone") {
+        return; // Allow access
+      }
+
+      // If view requires authentication but user is not logged in
+      if (!loggedIn.value) {
+        if (authStrategy === "auth0") {
+          return router.push("/login");
+        }
+        return; // For other auth strategies, allow access
+      }
+
+      // Check if user has sufficient role to access this view
+      if (loggedIn.value && user.value) {
+        const typedUser = user.value as User;
+        const userRole = typedUser.userRole || Role.Viewer;
+
+        // For 'member' permission, user must have Member or Admin role
+        if (permission === "member" && userRole < Role.Member) {
+          return router.push("/");
+        }
+
+        // For 'admin' permission, user must have Admin role
+        if (permission === "admin" && userRole < Role.Admin) {
+          return router.push("/");
+        }
+
+        // For 'signed-in' permission, any authenticated user can access
+        if (permission === "signed-in") {
+          return; // Allow access
+        }
+      }
```

Comment:
Light readability improvements

```suggestion
      // Public access: no login needed
      if (permission === "anyone") return;
      
      // Require authentication
      if (!loggedIn.value) {
        if (authStrategy === "auth0") return router.push("/login");
        return; // allow for other auth strategies
      }
      
      // Authenticated from here on
      const typedUser = user.value as User;
      const userRole = typedUser?.userRole ?? Role.Viewer;
      
      // Role-based access control
      switch (permission) {
        case "member":
          if (userRole < Role.Member) return router.push("/");
          break;
        case "admin":
          if (userRole < Role.Admin) return router.push("/");
          break;
        case "signed-in":
          return; // any signed-in user is fine
      }
```